Sources/Internal/SubviewsManager.swift

```diff
@@ -0,0 +1,105 @@
+// Created by Bryan Keller on 12/3/22.
+// Copyright © 2022 Airbnb Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+
+/// Manages adding item views to a parent view, ensuring that the subviews array remains sorted so that subviews have the correct
+/// order along the z-axis.
+final class SubviewsManager {
+
+  // MARK: Lifecycle
+
+  init(parentView: UIView) {
+    self.parentView = parentView
+  }
+
+  // MARK: Internal
+
+  // Only used for XCTest support
+  var _testSupport_insertedItemTypes = [VisibleItem.ItemType]()
```

Comment:
Hmm, would be nice to avoid this. What if this class's `insertSubview` returned the insertion index? Would that be sufficient for testing? Or this class could be simplified to not even manage the view's itself, but just manage indexes.

Reply:
Good call! Refactored in latest commit